### DISCUSSION TITLE
General questions.

### DISCUSSION THREAD

**Discussion by a4ginatl** [codefori/vscode-ibmi#1694] [Q&A] [answered]
I am watching Liam's videos on YouTube. The screens are different to what I have.

1. Do I have a different version?
2. The settings page is different. What I see is the generic stuff that comes with VSC. How do I get to the VSC settings?
3. In the video, there is a Member browser. I do not have it. How can I get it?
4. I also do not have the Database browser. I have loaded DB2 for i. How do I get it?
5. Liam shows spool files in the bottom panel. I cannot get to display this. I do not seem to have the option. I need to compile a program to get a listing in the terminal tab. 
6. Can I display spool files from the OUTQ's?

Thanks, Darryl Freinkel

<hr />

⚠️ **REMOVE THIS LINE AND ANY SENSITIVE INFORMATION BELOW!** ⚠️

|Context|Version|
|-|-|
|Code for IBM i version|2.4.0|
|Visual Studio Code version|1.84.2|
|Operating System|win32_x64|

<details>
<summary>Active extensions</summary>

```
Code Coverage for IBM i (code-coverage-ibmi): 0.1.7
Code for IBM i Walkthroughs (vscode-ibmi-walkthroughs): 0.4.0
Db2 for IBM i (vscode-db2i): 0.6.0
Emmet (emmet): 1.0.0
Error Lens (errorlens): 3.15.0
Git (git): 1.0.0
Git Base (git-base): 1.0.0
GitHub (github): 0.0.1
IBM i Notebooks (vscode-ibmi-notebooks): 0.0.6
JSON Language Features (json-language-features): 1.0.0
Merge Conflict (merge-conflict): 1.0.0
Node Debug Auto-attach (debug-auto-launch): 1.0.0
TODO Highlight (vscode-todo-highlight): 1.0.5
TypeScript and JavaScript Language Features (typescript-language-features): 1.0.0
WSL: Recommender (remote-wsl-recommender): 0.0.19
```
</details>

<hr />

<details>
<summary>Remote system</summary>

|Setting|Value|
|-|-|
|IBM i OS|V7R4M0|
|Tech Refresh|7|
|CCSID|65535|
|SQL|Disabled
|Source dates|Enabled

### Enabled features

|/usr/bin|/QOpenSys/pkgs/bin|/QSYS.lib/ILEDITOR.lib|/QSYS.LIB|/QIBM/ProdData/IBMiDebugService/bin|
|-|-|-|-|-|
|attr|bash|GENCMDXML.PGM|QZDFMDB2.PGM|startDebugService.sh|
|iconv|chsh|GETNEWLIBL.PGM|||
|setccsid|ls||||
|tar|md5sum||||
||sort||||
||stat||||
</details>

<details>
<summary>Shell env</summary>

```bash
BUILDLIB=QGPL
CURLIB=QGPL
HOME=/home/dfreinkel
HOST=172.17.72.21
LIBLS=DPOBJLIB DPPDATA DPSRCLIB LIBHTTP YAJL QTEMP QGPL OPRUTLLIB DFREINKEL
LOGIN=dfreinkel
LOGNAME=dfreinkel
MAIL=/var/spool/mail/dfreinkel
OLDPWD=/home/DFREINKEL
PASE_USRGRP_LIMITED=N
PATH=/QOpenSys/usr/bin:/usr/ccs/bin:/QOpenSys/usr/bin/X11:/usr/sbin:.:/usr/bin
PWD=/home/dfreinkel
SHELL=/QOpenSys/pkgs/bin/bash
SHLVL=1
SSH_CLIENT=10.27.230.56 52013 22
SSH_CONNECTION=10.27.230.56 52013 172.17.72.21 22
TZ=<EST>5<EDT>,M3.2.0,M11.1.0
USER=dfreinkel
USERNAME=dfreinkel
_=/QOpenSys/usr/bin/env
```
</details>

<details>
<summary>Variants</summary>

```json
{
  "american": "#@$",
  "local": "#@$"
}
```
</details>

<details>
<summary>Errors</summary>

```json
[
  {
    "command": "system \"CRTLIB LIB(ILEDITOR) TEXT('Code for i temporary objects. May be cleared.')\"",
    "code": 255,
    "stderr": "CPF2111: Library ILEDITOR already exists.",
    "cwd": "/home/dfreinkel"
  },
  {
    "command": "system \"DLTOBJ OBJ(ILEDITOR/O_*) OBJTYPE(*FILE)\"",
    "code": 255,
    "stderr": "CPF2125: No objects deleted.",
    "cwd": "/home/dfreinkel"
  }
]
```
</details>

**Comment by chrjorgensen** [comment]
@a4ginatl 

> Do I have a different version?

Yes, the Code for IBM i version available today is quite different from the version shown in the videos - many updates are released each year, and the look and behavior of the extension has changed a lot.

> The settings page is different. What I see is the generic stuff that comes with VSC. How do I get to the VSC settings?

Do you mean the settings for the Code for IBM i extension? These can be found in the VS Code settings panel (press `Ctrl-,`) under `Extensions` and `Code for IBM i`:

![image](https://github.com/codefori/vscode-ibmi/assets/13275072/aac86d53-3021-4b4f-bd8b-cc5ff7bf96a2)

To get to the settings for a server when not connected, right-click the server in the `SERVERS` list and select `Connections Settings`.
To get to the settings for a server when connected, click the server name in the status bar and select `Settings`.

![image](https://github.com/codefori/vscode-ibmi/assets/13275072/3ac56601-cf27-4b31-8372-d4d3f40b108c)

> In the video, there is a Member browser. I do not have it. How can I get it?

The Member browser has been replaced by the Object Browser.

> I also do not have the Database browser. I have loaded DB2 for i. How do I get it?

The Database browser has been replaced by the DB2 for i - so you've already got it!

> Liam shows spool files in the bottom panel. I cannot get to display this. I do not seem to have the option. I need to compile a program to get a listing in the terminal tab.

The output from Actions (compiles or other commands) is still found in the Terminal, but in a separate tab - or you can click the `Open output` in the action message to show the output in a new tab in the main area:

![image](https://github.com/codefori/vscode-ibmi/assets/13275072/5135c3e7-45fa-42b2-9104-f54c9b161510)

> Can I display spool files from the OUTQ's?

No, not yet - it may come in the add-on extension [vscode-ibmi-fs](https://github.com/codefori/vscode-ibmi-fs) later on...

**Comment by a4ginatl** [comment]
Christian, thank you.

 

Yes, the Code for IBM i version available today is quite different from the version shown in the videos - many updates are released each year, and the look and behavior of the extension has changed a lot.

The settings page is different. What I see is the generic stuff that comes with VSC. How do I get to the VSC settings?

Do you mean the settings for the Code for IBM i extension? These can be found in the VS Code settings panel (press Ctrl-,) under Extensions and Code for IBM i:

image.png (view on web) <https://github.com/codefori/vscode-ibmi/assets/13275072/aac86d53-3021-4b4f-bd8b-cc5ff7bf96a2> 

CTRL - = Zoom 

CTRL + = Zoom

I cannot find how to get to that page.

To get to the settings for a server when not connected, right-click the server in the SERVERS list and select Connections Settings.
To get to the settings for a server when connected, click the server name in the status bar and select Settings. Some options are not there.

image.png (view on web) <https://github.com/codefori/vscode-ibmi/assets/13275072/3ac56601-cf27-4b31-8372-d4d3f40b108c> 

As there have been so many improvements, could someone please do a video on the changes when they come out. Basically a change review. It does not need to be in detail but helps in cases where we start with older versions and I could not find anything about the improvements. Maybe its there, but I don’t know where to look.

Thank you.

Darryl Freinkel

A4G

Telephone: 770.321.8562 Mobile: 678.355.8562

**Comment by a4ginatl** [comment]
I have installed IBMi Notepad but it does not work. I am following Liam’s video and I cannot get it to work. There is nothing in the search bar when I type  in “Note”

 

It shows as installed.

 

What could be the problem?

Darryl Freinkel

A4G

Telephone: 770.321.8562

**Reply by sebjulliand** [reply, marked as answer]
The notepad has been moved to the DB2 for i extension but it's not been released yet, I think. The current Notepad extension is not maintained anymore and won't work in its current state.
